### DISCUSSION TITLE
Is there any ARGO CD integration plans ?

### DISCUSSION THREAD

**Discussion by helixphoenix** [kiali/kiali#3741] [General]
During istioconn we had common interest on ARGO CD integration. Therefore would like to follow the progress here.

**Comment by abonas** [comment]
Thanks. Do you have a specific feature in mind?

**Reply by helixphoenix** [reply]
Argo CD already has some visuals but for instance it would be nice if we could automatically add a  service/namespace to kiali when its deployed via argo cd.

**Reply by abonas** [reply]
I'm not sure I understand the ask, because Kiali is getting the services/namespaces from k8s/openshift along with more data from Istio via p8s, so as entities are added there, they will be reflected in Kiali. Or there is another feature that you refer to?

**Comment by lucasponce** [comment]
Hi @phoenitunes, 
In general, third party integrations in kiali as possible, and those are handled in the way of optional extensions
https://kiali.io/documentation/latest/extensions/

The comment "if we could automatically add a service/namespace to kiali when its deployed via argo cd." is not fully clear as Kiali doesn't operate in the deployment lifecycle, I mean when some actor deploys on Istio (using plain Kubernetes or any CI/CD tool) Kiali just get the kubernetes set of applications (workloads + services) and combine that info with all telemetry, signals and configuration collected by Istio.

So, it would be interesting to get more feedback about the highlevel use case you'd like to propose.

Specially on those aspects that are integrated with Istio which is the primary goal of Kiali.

Thanks for opening the discussion.

Extensions is an interesting topic.

**Comment by jshaughn** [comment]
I admittedly don't have any experience with Argo CD or GitOps but maybe what is being asked here is whether Kiali, or an extension, can help ensure desired changes to the Istio object definitions somehow get updated into the git definitions?  I think we need a user of GitOps and Istio to describe the pain points.
